src/container-mapping.ts

```diff
@@ -81,74 +80,104 @@ export class ContainerMapping implements IMicrosoftSecurityDevOps {
         }
         console.log(`PreJobStartTime: ${startTime}`);
 
+        let dockerVersion = [];
         let reportData = {
-            dockerVer: "",
-            dockerEvents: "",
-            dockerImages: ""
+            dockerVersion: "",
+            dockerEvents: [],
+            dockerImages: []
         };
-        
+
         // Initialize the commands 
-        await exec.exec('docker --version', null, {
-            listeners: {
-                stdout: (data: Buffer) => {
-                    reportData.dockerVer = reportData.dockerVer.concat(data.toString());
-                }
-            }
+        await this.execCommand('docker --version', dockerVersion)
+        .catch((error) => {
+            throw new Error("Unable to get docker version: " + error);
         });
-        await exec.exec(`docker events --since ${startTime} --until ${new Date().toISOString()} --filter event=push --filter type=image --format ID={{.ID}}`, null, {
-            listeners: {
-                stdout: (data: Buffer) => {
-                    reportData.dockerEvents = reportData.dockerEvents.concat(data.toString());
-                }
-            }
+        // The backend expects the docker version to be a string, not an array
+        reportData.dockerVersion = dockerVersion.join('');
+
+        await this.execCommand(`docker events --since ${startTime} --until ${new Date().toISOString()} --filter event=push --filter type=image --format ID={{.ID}}`, reportData.dockerEvents)
+        .catch((error) => {
+            throw new Error("Unable to get docker events: " + error);
         });
-        await exec.exec('docker images --format CreatedAt={{.CreatedAt}}::Repo={{.Repository}}::Tag={{.Tag}}::Digest={{.Digest}}', null, {
+
+        await this.execCommand(`docker images --format CreatedAt={{.CreatedAt}}::Repo={{.Repository}}::Tag={{.Tag}}::Digest={{.Digest}}`, reportData.dockerImages)
+        .catch((error) => {
+            throw new Error("Unable to get docker images: " + error);
+        });
+
+        core.debug("Finished data collection, starting API calls.");
+
+        let bearerToken: string | void = await core.getIDToken()
+            .then((token) => { return token; })
+            .catch((error) => {
+                throw new Error("Unable to get token: " + error);
+            });
+
+        if (!bearerToken) {
+            throw new Error("Empty OIDC token received");
+        }
+
+        var reportSent: boolean = await this.sendReport(JSON.stringify(reportData), bearerToken, sendReportRetryCount);
+        if (!reportSent) {
+            throw new Error("Unable to send report to backend service");
+        };
+    }
+
+    /**
+     * Execute command and setup the listener to capture the output
+     * @param command Command to execute
+     * @param listener Listener to capture the output
+     * @returns a Promise

```

Comment:
```suggestion
     * @returns a boolean Promise

```
nit (just copying below)

Reply:
This is not a boolean, it does not return anything